### PR TITLE
Update behavior of `inmanta module install`

### LINKED ISSUE #6717: Update behavior of `inmanta module install`
*Temporarily blocked on the outcome of [this Slack discussion](https://inmanta.slack.com/archives/CKRF0C8R3/p1699455924245889)*

as follow up to  https://github.com/inmanta/inmanta-core/pull/6655

Update the pip config used by `inmanta module install` to behave as follows:

1.  raise an error stating that the command is no longer supported.
2.  Offer the suggestion to run `pip install -e .` to install a module in development mode
3.  Offer the suggestion to run `inmanta module build` followed by `pip install` if the user really does not want an editable install.
4.  The `inmanta module build` command should use `use-system-config=true`. It is assumed safe because it only installs build dependencies.

# Tasks

- [ ] implement behavior as described above
- [ ] update docs

### PR DESCRIPTION
**Pull request by FloLey**
# Description

Update the pip config used by inmanta module install to behave as follows:
1. raise an error stating that the command is no longer supported.
2. Offer the suggestion to run pip install -e . to install a module in development mode
3. Offer the suggestion to run inmanta module build followed by pip install if the user really does not want an editable install.
4. The inmanta module build command should use use-system-config=true. It is assumed safe because it only installs build dependencies.

closes #6717 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: tests/test_module_loader.py [hunk 1]
```diff
@@ -616,10 +615,11 @@ def test_project_requirements_dont_overwrite_core_requirements_source(
     jinja2_version_before = active_env.get_installed_packages()["Jinja2"].base_version
 
     # Install the module
-    with pytest.raises(InvalidModuleException) as e:
-        ModuleTool().install(editable=False, path=module_path)
+    with pytest.raises(ConflictingRequirements) as e:
+        ModuleTool().build(path=module_path)
+        env.process_env.install_from_source([env.LocalPackagePath(path=module_path, editable=False)])
```
Comment (Hugo-Inmanta) [inline tests/test_module_loader.py:620]: I think only the `install_from_source` is expected to raise a `ConflictingRequirements` ? If so, I would move the `build` part outside of the `with` block to make it clearer

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/moduletool.py [hunk 2]
```diff
@@ -1614,7 +1591,7 @@ def __init__(self, module_path: str) -> None:
 
     def build(self, output_directory: str, dev_build: bool = False, byte_code: bool = False) -> str:
         """
-        Build the module and return the path to the build artifact.
+        Build the module using the pip system config and return the path to the build artifact.
```
Comment (Hugo-Inmanta) [inline src/inmanta/moduletool.py:1594]: I can't find where the system's pip config is being used during the build, is this already implemented ?

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

File: src/inmanta/moduletool.py (hunk 2 above)
Comment (Hugo-Inmanta) [reply src/inmanta/moduletool.py:1594]: Ohh I see :+1:

File: tests/moduletool/test_install.py [hunk 3]
```diff
@@ -206,8 +215,7 @@ def test_dev_checkout(git_modules_dir, modules_repo):
 
 
 @pytest.mark.parametrize_any("editable", [True, False])
-@pytest.mark.parametrize_any("set_path_argument", [True, False])
-def test_module_install(snippetcompiler_clean, modules_v2_dir: str, editable: bool, set_path_argument: bool) -> None:
+def test_module_install(snippetcompiler_clean, modules_v2_dir: str, editable: bool) -> None:
```
Comment (arnaudsjs) [inline tests/moduletool/test_install.py:218]: The docstring still references the `inmanta module install` command.

File: tests/moduletool/test_install.py [hunk 4]
```diff
@@ -259,10 +267,13 @@ def test_module_install_conflicting_requirements(tmpdir: py.path.local, snippetc
         new_requirements=[InmantaModuleRequirement.parse(name) for name in ("modone", "modtwo")],
     )
 
-    with pytest.raises(module.InvalidModuleException) as exc_info:
-        run_module_install(module_path, False, True)
-    assert isinstance(exc_info.value.__cause__, ConflictingRequirements)
-    assert "caused by:" in exc_info.value.format_trace()
+    with pytest.raises(ConflictingRequirements) as exc_info:
```
Comment (arnaudsjs) [inline tests/moduletool/test_install.py:270]: Docstring still references the install command.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/6717-update-module-install.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+description: Removed the 'inmanta module install' command.
+change-type: major
+issue-nr: 6717
+destination-branches: [master]
+sections:
+  deprecation-note: "{{description}}. As an alternative to the now-removed 'inmanta module install' command,
+  users should follow the updated procedure for module installation:
+  The new method involves using the 'inmanta module build' command followed by 'pip install ./dist/<dist-package>' to
+  build a module from source and install the distribution package, respectively."
```
Comment (arnaudsjs) [inline changelogs/unreleased/6717-update-module-install.yml:9]: The command for an editable install is missing here.

**Review by Hugo-Inmanta — APPROVED** [review]

File: tests/moduletool/test_install.py [hunk 6]
```diff
@@ -206,11 +215,9 @@ def test_dev_checkout(git_modules_dir, modules_repo):
 
 
 @pytest.mark.parametrize_any("editable", [True, False])
-@pytest.mark.parametrize_any("set_path_argument", [True, False])
-def test_module_install(snippetcompiler_clean, modules_v2_dir: str, editable: bool, set_path_argument: bool) -> None:
+def test_module_install(snippetcompiler_clean, modules_v2_dir: str, editable: bool) -> None:
     """
-    Install a simple v2 module with the `inmanta module install` command. Make sure the command works with all possible values
-    for its options.
+    Make sur it is possible to install a module in both non-editable and editable mode
```
Comment (arnaudsjs) [inline tests/moduletool/test_install.py:220]:
```suggestion
    Make sure it is possible to install a module in both non-editable and editable mode
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**